mayastor/src/lib.rs

```diff
@@ -58,6 +65,69 @@ pub extern "C" fn cps_init() {
     bdev::nexus::register_module();
 }
 
+extern "C" fn log_impl(
+    level: i32,
+    file: *const c_char,
+    line: u32,
+    _func: *const c_char,
+    buf: *const c_char,
+    _n: i32, // the number of bytes written into buf
+) {
+    unsafe {
+        let fmt = CStr::from_ptr(buf).to_str().unwrap().trim_end();
+        let filename = CStr::from_ptr(file).to_str().unwrap();
+
+        if spdk_log_get_print_level() < level {
+            return;
+        }
+
+        let lvl = match level {
+            -1 => return,
+            0 => Level::Error,
+            1 => Level::Warn,
+            2 => Level::Info,
+            3 => Level::Debug,
+            4 => Level::Trace,
+            _ => Level::Error,
```

Comment:
IMHO it would have been better to match levels by SPDK enum names rather than numbers (i.e. spdk_sys::SPDK_LOG_WARN instead of 1).